lib/rmt/cli/custom_repos.rb

```diff
@@ -0,0 +1,90 @@
+class RMT::CLI::CustomRepos < RMT::CLI::Base
+
+  include ::RMT::CLI::RepoPrintable
+
+  desc 'add', 'Adds a custom repository to a product'
+  option :name, aliases: '-n', type: :string, desc: 'The name of the custom repository', required: true
+  option :url, aliases: '-u', type: :string, desc: 'Absolute external URL to this repository', required: true
+  option :update, type: :boolean, desc: 'Update repository instead of ignore when it already exists', default: false
+  option :product_id, aliases: '-p', type: :string, required: true,
+         desc: 'The id of the product where the repository should be added. Use `rmt-cli products list` to see products'
+  def add
```

Comment:
Your options name, url and product_id are not really optional. They are all required. For this kind of data, we should use arguments, not options. This also makes the command shorter to type.
So please change to `def add(name, url, product_id)` and remove the options.
Also, let's think if we really need the last remaining option "update yes/no" ... I don't see its value. You could solve it by just always updating. Maybe add a user confirmation. Or block updating and output a warning, that a repo with this url already exists, plus the hint to just delete that one, in case someone wants to "update" it... Less options make it easier to test and easier to understand for users. IMO.